lms/djangoapps/django_comment_client/management/commands/tests/test_rename_user.py

```diff
@@ -0,0 +1,316 @@
+                            }
+                        }
+                    ],
+                    "username" : cls.sql_users[1].username
+                }
+
+        user_3 = {
+                    "default_sort_key" : "date",
+                    "notification_ids" : [ ],
+                    "external_id" : "test_user_3",
+                    "username" : cls.sql_users[2].username,
+                    "email" : "robot3@robots.com"
+                }
+
+        cls.mongo_user_ids.append(cls.db.users.insert(user_1))
+        cls.mongo_user_ids.append(cls.db.users.insert(user_2))
+        cls.mongo_user_ids.append(cls.db.users.insert(user_3))
+
+
+
+    def test_mongo_seed(self):
+        self.assertTrue(len(self.mongo_user_ids) == 3, 'Test mongo users were not properly seeded, rendering tests invalid.')
+        self.assertTrue(len(self.mongo_comment_ids) == 3, 'Test mongo comments were not properly seeded, rendering the tests invalid.')
+
+    def test_incorrect_num_args(self):
+        zero_args = []
+        one_arg = ['arg']
+        three_args = ['arg', 'arg', 'arg']
+        self.assertRaises(CommandError, self.rename_cmd.handle, *zero_args)
+        self.assertRaises(CommandError, self.rename_cmd.handle, *one_arg)
+        self.assertRaises(CommandError, self.rename_cmd.handle, *three_args)
+
+    def test_user_does_not_exist(self):
+        args = ['---@---.com', '---2@---2.com']
+        self.assertRaises(LookupError, self.rename_cmd.handle, *args)
+
+
+    def test_change_username_basic(self):
+        #ensure that the starting sql username is different from the target username
+        self.assertTrue(self.sql_users[0].username != TestRenameUser.NEW_USERNAME)
+
+        #ensure that the starting mongo username matches the starting sql username
+        mongo_user = self.db.users.find_one({'username':self.sql_users[0].username})
+        self.assertTrue(mongo_user['username'] == self.sql_users[0].username)
+        mongo_comment = self.db.contents.find_one({'author_username':self.sql_users[0].username})
+        self.assertTrue(mongo_comment['author_username'] == self.sql_users[0].username)
+
+        #perform the rename
+        args = [self.sql_users[0].username, TestRenameUser.NEW_USERNAME]
+        self.rename_cmd.handle(*args)
+
+        #check that the sql username now matches the target username
+        self.sql_users[0] = User.objects.get(id=self.sql_users[0].id)
+        self.assertEqual(self.sql_users[0].username, TestRenameUser.NEW_USERNAME)
+
+        #check that the mongo username now matches the target username
+        mongo_user = self.db.users.find_one({'_id':mongo_user['_id']})
+        self.assertEqual(mongo_user['username'], TestRenameUser.NEW_USERNAME)
+
+        mongo_comment = self.db.contents.find_one({'_id':mongo_comment['_id']})
+        self.assertEqual(mongo_comment['author_username'], TestRenameUser.NEW_USERNAME)
+
+
+
+    def test_change_username_advanced(self):
```

Comment:
It seems that the behavior you're trying to test for here is that the name change has no effect on an existent user whose username doesn't match the 'before" input.  Maybe you can simplify this test down to just capture that (probably only need to check `sql_users[1]`, since the above test really already checks the behavior of `sql_user[0]` and `[2]`.

Also you might name the test something like `test_change_username_non_matching` or somesuch